### ISSUE TITLE
Property 'getAttrs' in type 'BaseClientContext' is not assignable to the same property in base type 'ClientContext'

### ISSUE THREAD

**Issue by matt-oplo** [featurehub-io/featurehub-javascript-sdk#151]
**Describe the bug**
Using the SDK in a Typescript project with strict type checking enabled, produces the following error:
```
node_modules/featurehub-javascript-client-sdk/dist/context_impl.d.ts:24:5 - error TS2416: Property 'getAttrs' in type 'BaseClientContext' is not assignable to the same property in base type 'ClientContext'.
  Type '(key: string) => string[] | undefined' is not assignable to type '(key: string) => string[]'.
    Type 'string[] | undefined' is not assignable to type 'string[]'.
      Type 'undefined' is not assignable to type 'string[]'.

24     getAttrs(key: string): Array<string> | undefined;
       ~~~~~~~~
```
This is because the return types in the `BaseClientContext` abstract class and the `ClientContext` interface are different.

**BaseClientContext:**
```
getAttrs(key: string): Array<string> | undefined;
```
**ClientContext:**
```
getAttrs(key: string): Array<string>;
```

 Which area does this issue belong to?

* [x] SDK
* [ ] SDK examples
* [ ] Documentation
* [ ] Other

**To Reproduce**
1. Create a new typescript project.
2. Set `"strict": true` in `tsconfig.json`
3. Create a file that imports FeatureHubConfig.
4. Build (tsc)
5. See error

**Expected behavior**
Project builds without errors.

**Versions**
 - SDK version [1.2.0]
 - OS: [Ubuntu 22.04]
 - NodeJS: [v16.18.0]
 - Typescript: ["ts-loader": "^9.4.2", "ts-node": "^10.9.1", "typescript": "^4.9.5"]

**Comment by rvowles** [comment]
Now that opened a can of worms 😁

**Comment by rvowles** [comment]
@matt-oplo there is an attached PR, not sure if you would like to pull it down and compile it (if you use npm run compile - it will build and link it, so you can `npm link featurehub-javascript-client-sdk` into your codebase and check)

**Comment by matt-oplo** [comment]
I've pulled the PR, built it, and linked it to my project which now builds without error.

**Comment by rvowles** [comment]
done and released as 1.3.0 @matt-oplo

**Comment by matt-oplo** [comment]
Thanks for your work on this @rvowles
